presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcPageSource.java

```diff
@@ -82,7 +85,8 @@ public OrcPageSource(
             List<Long> columnIds,
             List<Type> columnTypes,
             List<Integer> columnIndexes,
-            UUID shardUuid)
+            UUID shardUuid,
+            AggregatedMemoryContext systemMemoryUsage)
```

Comment:
Instead of having this context which is shared with the `OrcRecordReader` why not just add a `getSystemMemoryUsage` to the reader and return that in `getSystemMemoryUsage`?